index.html

```diff
@@ -0,0 +1,255 @@
+
+```graphql
+#::[Using the `@using` directive]
+schema @using(spec: "https://spec.example.com/example/v1") {
+  query: Query
+}
+```
+
+### `spec`
+
+Provide the specification URL as a string.
+
+The `spec` argument **should** be the URL of a specification. When viewed, the URL **should** provide the content of the appropriate version of the specification in some human-readable form. In short, a human reader should be able to click the link and go to the docs. It is not required that the content be machine-readable in any particular way.
+
+Processors **must** degrade gracefully. Specifically, if the `spec` argument is not a [valid RFC 3986 URL](https://tools.ietf.org/html/rfc3986), processors **must** treat it as an opaque, non-versioned identifier referencing a particular specification.
+
+The final component of the URL's `"/"`-separated `path` **must** be a valid [SemVer specifier](#semver-specifiers) identifying the version of the spec in use. Everything prior to the version specifier is the spec's *base URL*.
+
+Specifications **should** prefix any schema elements they introduce. The final component of the *base URL*'s `"/"`-separated `path` **should** be the default prefix under which new schema elements will be introduced. For example:
+
+```graphql
+#::[Using the `@using` directive with a default prefix]
+#                    |---------- 👇🏽 base url --------|👇🏽 version
+schema @using(spec: "https://spec.example.com/example/v1") {
+#                                        prefix ☝🏾
+  query: Query
+}
+
+type User {
+               # prefixed directive from the example spec
+  name: String @example_someDirective
+}
+```
+
+Processors **should** otherwise treat the spec's base URL as an opaque string identifying the specification.
+
+This spec assigns no meaning to additional URL components, such as the query string or fragment. Processors **should** ignore any meaningless URL components when parsing the base URL and semver specifier.
+
+### `prefix`
+
+Change the prefix for schema elements from this specification.
+
+```graphql
+#::[Using the `@using` directive with a custom prefix]
+schema @using(spec: "https://spec.example.com/example/v1", prefix: "eg") {
+  query: Query
+}
+
+type User {
+  name: String @eg_someDirective
+}
+```
+
+### `for`
+
+Describe the purpose of a spec within a document.
+
+```graphql
+schema @using(spec: "https://example.com/spec/policy/v1",
+               for: ["execution", "security"]) {
+  ...
+}
+```
+
+Documents reference specifications for a variety of purposes: to provide documentation, security policies, routing configuration, and so on. This argument describes *why* a spec is being imported. Consumers **may** choose to reject schemas which use unknown specs `for` important purposes. For example, a graph router may decide to fail if presented with a schema that uses an unknown spec `for: ["security"]`.
```

Comment:
@queerviolet I'm not opposed to the `for` argument as it's described here, but I know we've been thinking about using it to determine the behavior of the gateway when/if a directive defined by the spec has no known implementation in the gateway.

Here's my concern with that plan: what if the spec declares multiple different directives that should have different fallback behaviors (some can be ignored, some might deserve a warning, but others should trigger a hard failure)? Wouldn't we need a finer-grained configuration syntax for those individual directives?

If the answer is to split the spec up, that introduces the possibility of independent versions for multiple specs, along with multiple `@using` declarations, which might not be desirable for a spec that intentionally groups multiple declarations together using a single version.

Straw-dog sketch:
```graphql
schema @using(spec: "https://example.com/spec/policy/v1",
              for: ["execution", "security"],
              fail_if_unimplemented: ["@auth"],
              warn_if_unimplemented: ["@detach"]) {
  ...
}
```
This would provide a different fallback behavior for `@policy_auth` than for `@policy_detach`, though the spec defines both `@auth` and `@detach` in one document (imagine that's desirable for whatever reason). Any other directives would simply be ignored if unimplemented.

This information would be passed along to the composer, and would remain in the final CSDL, so whatever system is responsible for serving this composed graph can verify the implementations of any `fail_if_unimplemented` or `warn_if_unimplemented` directives, and react appropriately.

To complicate the syntax further, it seems conceivable that these fallback behaviors might depend somehow on the `for` context, so perhaps these argument APIs could/should be combined? Not sure how fancy the API really _needs_ to be in this version of the `@using` spec, but it might be worth thinking a step ahead to be sure we can adapt/expand the syntax if the need arises.

Reply:
In theory, the spec itself could also annotate its important directives with these rules, but I haven't been able to find a syntactic place to add that information to a [directive declaration](https://spec.graphql.org/draft/#sec-Type-System.Directives), other than possibly using some special naming convention for the directive, or abusing its argument declarations (which [can have their own directives](https://spec.graphql.org/draft/#sec-Field-Arguments), but that seems strange for metadata that applies to the whole directive).

Reply:
One question is: who should fail if unimplemented? should explorer fail if it doesn't understand the `@policy_auth` directive? that was the motivation behind the more declarative approach of specifying the spec role.

You can handwave it and say well obviously explorer should never fail in that case, but if you fast forward a few years and explorer is now doing composition internally and powering half the world's dashboards, and it starts to seem potentially dodgier.

Re: versions, `@using` is already repeatable, so we already have the problem of multiple versions of multiple dependent/independent specs, which yes, could be tricky—although, i'll note, no trickier than the current situation of having multiple versions of multiple specs, only with no indication of what version of which specs are in use. That seems worse.

More conceptually, once you even kindof implement a spec, my gut feeling is that we (the `@using` writers, specifically) do not care about you anymore. Specs can define their own validations. The question here is what to do if we've never even heard of a particular spec.

Re: annotating directives—I too would like directives on directives for other reasons, but we don't have them, and I don't see how they're helpful here? If an implementation is available, it should conform to the spec, which can include language in human words, like "gateways which do not support auth should reject any schemas which contain `@auth` directives."

Reply:
`for` is gone